frontend/javascripts/oxalis/model/sagas/annotation_saga.ts

```diff
@@ -159,4 +178,75 @@ export function* watchAnnotationAsync(): Saga<void> {
   );
   yield* takeLatest("EDIT_ANNOTATION_LAYER", pushAnnotationLayerUpdateAsync);
 }
-export default [warnAboutSegmentationZoom, watchAnnotationAsync];
+
+export function* acquireAnnotationMutexMaybe(): Saga<void> {
+  yield* take("WK_READY");
+  const allowUpdate = yield* select((state) => state.tracing.restrictions.allowUpdate);
+  const annotationId = yield* select((storeState) => storeState.tracing.annotationId);
+  if (!allowUpdate) {
+    return;
+  }
+  const othersMayEdit = yield* select((state) => state.tracing.othersMayEdit);
+  const ONE_MINUTE = 1000 * 60;
+  let isInitialRequest = true;
+  let doesHaveMutex = false;
+  let shallTryAcquireMutex = othersMayEdit;
+
+  function onMutexStateChanged(canEdit: boolean, blockedByUser: APIUserCompact | null | undefined) {
+    if (canEdit) {
+      Toast.close("MutexCouldNotBeAcquired");
+      if (!isInitialRequest) {
+        Toast.success(messages["annotation.acquiringMutexSucceeded"]);
+        location.reload();
+      }
+    } else {
+      const message =
+        blockedByUser != null
+          ? messages["annotation.acquiringMutexFailed"]({
+              userName: `${blockedByUser.firstName} ${blockedByUser.lastName}`,
+            })
+          : messages["annotation.acquiringMutexFailed.noUser"];
+      Toast.error(message, { sticky: true, key: "MutexCouldNotBeAcquired" });
+    }
+  }
+
+  function* tryAcquireMutex(): Saga<void> {
+    while (shallTryAcquireMutex) {
+      if (!doesHaveMutex) {
+        yield* put(setAnnotationAllowUpdateAction(false));
+      }
+      try {
+        const { canEdit, blockedByUser } = yield* call(acquireAnnotationMutex, annotationId);
+        yield* put(setAnnotationAllowUpdateAction(canEdit));
+        if (canEdit !== doesHaveMutex || isInitialRequest) {
+          doesHaveMutex = canEdit;
+          onMutexStateChanged(canEdit, blockedByUser);
+        }
+      } catch (error) {
+        console.error("Error while trying to acquire mutex.", error);
+        if (doesHaveMutex === true || isInitialRequest) {
+          onMutexStateChanged(false, null);
+          yield* put(setAnnotationAllowUpdateAction(false));
```

Comment:
if the mutex acquisition fails because i'm offline, I'll immediately lose it. how about remembering the timestamp of the last acquisition and comparing it to the timeout constant? So, if I acquired the mutex one minute ago, I should still have it, right? maybe just add a retry around the acquisition request?

Reply:
also, with the current behavior I would have expected that the catch clause sets `doesHaveMutex` to false.

Reply:
Good catch. I just changed to code so that it retries up to one minute and that `doesHaveMutex` will be set to false